### DISCUSSION TITLE
2FA Authentication Problems

### DISCUSSION THREAD

**Discussion by AotrsCommander** [community/community#138431] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

The upcomign 2FA mandate is causing me some serious problems.

I do not have, nor can I afford to get, a smart phone or a security key.

I tried using the SMS authorisation, but it simply did not seem to send me a text message, despite trying twice (both omitting and putting in the leading 0), and waiting for a good half to three-quarters of an hour. This is kind of a problem.

The other linked suggestions in the 2FA page are either subscriptions services, or services which require a smartphone (i.e. googleplay or apple only).

I have yet to find a desktop authenticator that works with Windows, is also not a subscription cost (definitely not something I either can afford, nor am prepared to pay for given my usage of github is minimal (one mod and a fork of a package to print wargames record sheets) and works with Waterfox. (I have asked the Waterfox community if they can suggest one.)

I am beginning to suspect I will simply not be ABLE to get 2FA autherntication, especially if the SMS system is unusable. (When I have to use it for Amazon or Paypal, I usually have to request the code sent twice before my phone will even pick it up, something I don't seem to have the option to do here). 

This feels like the 2FA implimentation has not been considered very well; in having to do the "search for similiar discussions2 it was very clear this is a raft of problems.

This is the only place I seem to have even a remote chance to reaching out to the github support team (or even registering this is a problem). I am not even sure this is in th right place, but it's my best guess. (I was tempted to post this in Accessibility, as it feels dangerously being close to being "Get Out, Poor People," but resisted the urge on the basis that Being Broke is not yet considered a disability in its own right.)

If anyone can offer any suggestions (short of "become less broke") - ideally either how to reach the github team or something else that I can try, I would be very grateful.

**Comment by davevad93** [comment, marked as answer]
Hi @AotrsCommander ,

First of all,  if you need to contact support you can do it by opening a ticket [here](https://support.github.com/contact/account).

Regarding the 2FA thing,  with the Authenticator App option, you should be able to use 2FA with a TOTP browser extension that generates the codes directly in your browser. I don't use it as an option so I can't recommend any specific extension, but I'm pretty sure there are free ones, try searching for one and see if it works with GitHub.

![00000](https://github.com/user-attachments/assets/889dd25c-0548-4c6e-bf77-fc7bdd9470b8)

**Reply by AotrsCommander** [reply]
Thank you. I will submit a ticket (since they at least need to know that trhe SMS system is not 100% reliable). I asked on the Waterfox forums, and they were able to suggest KeePassXC (for anyone finding this sort of problem in the future) which seems sufficient.

**Reply by davevad93** [reply]
You're welcome @AotrsCommander 👍

**Comment by jgschmitz** [comment]
Some browser extensions, like Authy (which has a desktop app as well), might help you bypass the need for a smartphone. It's free, though not every extension will work with Waterfox, but it's worth checking if any desktop-based alternatives could work for you - also another option is WinAuth I haven't used it but its an open-source application that can generate time-based one-time passwords (TOTP) without needing a smartphone. https://winauth.github.io/winauth/download.html

**Comment by KuroodoD** [comment]
I also recommend using GitLab which doesn't have a ridiculous forced 2FA requirement

**Reply by AotrsCommander** [reply]
I will bear that in mind for future reference. (I have so little on here (all the more reason this is patently riduclous the amounr of time I've had to spend) that it would be even more time wasted to move it around now.)
